src/components/Comments/ViewComments/ViewComments.js

```diff
@@ -0,0 +1,124 @@
+
+const preparePayload = (method, headers, params) => {
+//   const url = `https://${domain}${path}`;
+  const url = 'http://localhost:8081/api/v1/post/viewcomments';
+  return {
+    method,
+    url,
+    headers,
+    params,
+  };
+};
+
+export const ViewComments = () => {
+  const classes = useStyles();
+  const [{ comments, page }, dispatch] = useStateValue();
+  const [errorInfo, setErrorInfo] = useState({ isError: false, errorMsg: null });
+
+  const urls = window.location.href;
+  const postId = urls.slice(urls.lastIndexOf('/') + 1, urls.length);
+  const token = cookie.load('jwtToken');
+  const commentHeaders = { Authorization: token };
+
+  const handleSuccessData = (data) => {
+    const { comments: newComments, totalPages } = data;
+    // setIsLoad(true);
+    dispatch({
+      type: 'changeComment',
+      newComments,
+    });
+    dispatch({
+      type: 'changePage',
+      newPage: { totalPages },
+    });
+    console.log(data);
+    console.log(page);
+  };
+
+  const fetchAllComments = async () => {
+    try {
+      const requestPayload = preparePayload('get', commentHeaders,
+        { postID: postId, page: page.currentPage, pageSize: page.pageSize });
+      const response = await serviceRequest(requestPayload);
+      if (response.status && response.status === 'success') {
+        const { data } = response;
+        handleSuccessData(data);
+      } else {
+        throw new Error('Internal Service Error');
+      }
+    } catch (err) {
+      const newErrorInfo = {
+        isError: true,
+        errorMsg: 'Internal Service Error',
+      };
+      setErrorInfo(newErrorInfo);
+    }
+  };
+
+  useEffect(() => {
+    fetchAllComments();
+  }, [page.currentPage, page.pageSize]);
+
+
+  return (
+    <div>
+      {errorInfo.isError && (<ErrorMessage message={errorInfo.errorMsg} styles={{ color: 'red' }} />)}
+      <List>
+        <div className={classes.commentHeader}>COMMENTS</div>
+        <hr />
+        {page.totalPages > 0
+          ? (
+            <List>
+              {comments.map((comment) => (
+                <ListItem key={comment.commentID} className={classes.list}>
+                  <ListItemAvatar>
+                    <Avatar
+                      alt="profile"
+                      src={`http://api.adorable.io/avatar/50/${comment.username}`}
+                    />
+                  </ListItemAvatar>
+                  <ListItemText
+                    primary={(
+                      <>
+                        <Typography
+                          component="span"
+                          className={classes.left}
+                          color="textPrimary"
+                        >
+                          {comment.username}
+                        </Typography>
+                        <Typography component="span" className={classes.right}>
+                          {comment.createdAt.substr(0, comment.createdAt.indexOf('T'))}
+                        </Typography>
```

Comment:
it is a good practice to extract the value and store into a variable first before return statement